### PR TITLE
fix be for business division create and list

### PR DESCRIPTION
**Pull request by ian-bpoc**
# Pull Request Review Comments

## Overview
- **Purpose of the PR**:
    - [/ ] Bug fix
    - [/ ] Refactoring

## Specific Areas for Verification
- Please specify any particular requests or questions for the reviewer here.

## Suggestions/Concerns
- If you have any suggestions or concerns, please mention them here.

---

_**Additional Comments for the Reviewer (Optional)**_

- I temporarily add the alert text for the duplicated entry of business name

### CODE REVIEW

File: client/src/pages/BusinessDivisions/BusinessDivisionsListAndEdit.tsx [hunk 1]
```diff
@@ -106,9 +102,11 @@ const BusinessDivisionsListAndEdit: React.FC = () => {
       }
 
       try {
+        const response = await axios.put('http://127.0.0.1:8000/api/master-business-division/bulk-update/', business, {
         // const response = await axios.put('http://127.0.0.1:8000/api/projectdatalist/update/', changes, {
-        const response = await axios.put('http://54.178.202.58:8000/api/projectdatalist/update/',  changes ,{
+        // const response = await axios.put('http://54.178.202.58:8000/api/projectdatalist/update/',  changes ,{
```
Comment (edward-stalley-work) [inline client/src/pages/BusinessDivisions/BusinessDivisionsListAndEdit.tsx:107]: Edit seems to not be working right now. Always gives duplicate error.




Screenshot 2024-09-30 at 12 27 17

**Review by edward-stalley-work — COMMENTED** [review]
Thank you Ian please see comment!

File: client/src/pages/BusinessDivisions/BusinessDivisionsListAndEdit.tsx (hunk 1 above)
Comment (ian-bpoc) [reply client/src/pages/BusinessDivisions/BusinessDivisionsListAndEdit.tsx:107]: I will take a look on it Ed

File: client/src/app/store.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import tableSlice from "../reducers/table/tableSlice";
 import graphSlice from "../reducers/graph/graphSlice";
 import personnelExpensesSlice from "../reducers/personnel/personnelExpensesSlice";
 import personnelSlice from "../reducers/personnel/personnelSlice";
-import businessDivisionsSlice from "../reducers/businessDivisions/businessDivisionsSlice";
+import businessDivisionsSlice from "../reducers/businessdivisions/businessdivisionsSlice";
```
Comment (edward-stalley-work) [inline client/src/app/store.ts:9]: This should be camelCase.
`businessDivisions/businessDivisionsSlice`

File: client/src/pages/Employees/EmployeesListAndEdit.tsx [hunk 3]
```diff
@@ -9,7 +9,7 @@ import AlertModal from "../../components/AlertModal/AlertModal";
 import { RiDeleteBin6Fill } from "react-icons/ri";
 import ListButtons from "../../components/ListButtons/ListButtons";
 import HeaderButtons from "../../components/HeaderButtons/HeaderButtons";
-import { fetchBusinessDivisions } from "../../reducers/businessDivisions/businessDivisionsSlice";
+import { fetchBusinessDivisions } from "../../reducers/businessdivisions/businessdivisionsSlice";
```
Comment (edward-stalley-work) [inline client/src/pages/Employees/EmployeesListAndEdit.tsx:12]: This should be camelCase.
businessDivisions/businessDivisionsSlice

File: client/src/pages/Employees/EmployeesRegistration.tsx [hunk 4]
```diff
@@ -6,7 +6,7 @@ import { useLocation, useNavigate } from 'react-router-dom'
 import Sidebar from '../../components/Sidebar/Sidebar'
 import RegistrationButtons from '../../components/RegistrationButtons/RegistrationButtons'
 import HeaderButtons from '../../components/HeaderButtons/HeaderButtons'
-import { fetchBusinessDivisions } from '../../reducers/businessDivisions/businessDivisionsSlice'
+import { fetchBusinessDivisions } from '../../reducers/businessdivisions/businessdivisionsSlice'
```
Comment (edward-stalley-work) [inline client/src/pages/Employees/EmployeesRegistration.tsx:9]: This should be camelCase.
businessDivisions/businessDivisionsSlice

File: client/src/pages/BusinessDivisions/BusinessDivisionsRegistration.tsx [hunk 5]
```diff
@@ -54,66 +60,111 @@ const BusinessDivisionsRegistration = () => {
...
+          ])
+        }
+        catch (error)
+        {
+          if (error.response && error.response.status === 401) {
+            console.error('Validation error:', error.response.data);
+            window.location.href = '/login'
+          } else {
+            console.error('There was an error creating the project planning data!', error)
+            alert('Business Name Already Exist')
```
Comment (edward-stalley-work) [inline client/src/pages/BusinessDivisions/BusinessDivisionsRegistration.tsx:139]: Please add an 's' and remove capitals except for first 'B'.

`Business name already exists`

File: client/src/pages/BusinessDivisions/BusinessDivisionsListAndEdit.tsx [hunk 6]
```diff
@@ -258,40 +322,41 @@ const BusinessDivisionsListAndEdit: React.FC = () => {
...
-                                  <td className='BusinessDivisionsListAndEdit_table_body_content_vertical'></td>
+                                  <td className='BusinessDivisionsListAndEdit_table_body_content_vertical'>{userMap[business_data.auth_user_id] || 'Unknown User'}</td>
+                                  <td className='BusinessDivisionsListAndEdit_table_body_content_vertical'>{formatDate(business_data.created_at)}</td>
+                                  <td className='BusinessDivisionsListAndEdit_table_body_content_vertical'>{formatDate(business_data.updated_at)}</td>
                                   <td className='BusinessDivisionsListAndEdit_table_body_content_vertical'>
                                     {/* <RiDeleteBin6Fill className='delete-icon' onClick={() => openModal('project')}/> */}
                                   </td>
                                 </tr>
-                              {/* ))} */}
+                              ))}
```
Comment (edward-stalley-work) [inline client/src/pages/BusinessDivisions/BusinessDivisionsListAndEdit.tsx:359]: No delete functionality right now. Please add a trash bin and delete like the other Edit pages!

Screenshot 2024-10-01 at 11 28 30

Example from Employees
Screenshot 2024-10-01 at 11 28 59

**Review by edward-stalley-work — CHANGES_REQUESTED** [review]
Thank you Ian.
Please see comments.

Summary:
When I opened your branch I had a lot of BusinessDivisionSlice errors.
This is due to lowercase instead of camelcase.

Also clients does not work on your branch so you need to pull the latest changes from develop and deal with any conflicts on your end.

Finally please add delete button/function!

File: client/src/app/store.ts (hunk 2 above)
Comment (ian-bpoc) [reply client/src/app/store.ts:9]: already fix the path into CamelCase and revert current changes

File: client/src/pages/Employees/EmployeesListAndEdit.tsx (hunk 3 above)
Comment (ian-bpoc) [reply client/src/pages/Employees/EmployeesListAndEdit.tsx:12]: already fix the path into CamelCase and revert current changes

File: client/src/pages/Employees/EmployeesRegistration.tsx (hunk 4 above)
Comment (ian-bpoc) [reply client/src/pages/Employees/EmployeesRegistration.tsx:9]: already fix the path into CamelCase and revert current changes

File: client/src/pages/BusinessDivisions/BusinessDivisionsRegistration.tsx (hunk 5 above)
Comment (ian-bpoc) [reply client/src/pages/BusinessDivisions/BusinessDivisionsRegistration.tsx]: included on my new commit https://github.com/carrieayu/bop/pull/152/commits/bcabfd06b2ab4ef03072c8a5e806f396591958cb together with reverting the file format

File: client/src/pages/BusinessDivisions/BusinessDivisionsListAndEdit.tsx (hunk 6 above)
Comment (ian-bpoc) [reply client/src/pages/BusinessDivisions/BusinessDivisionsListAndEdit.tsx:383]: added commit on delete function

File: client/src/pages/BusinessDivisions/BusinessDivisionsListAndEdit.tsx [hunk 7]
```diff
@@ -118,10 +118,10 @@ const BusinessDivisionsListAndEdit: React.FC = () => {
...
       try {
         const response = await axios.put('http://127.0.0.1:8000/api/master-business-division/bulk-update/', business, {
-          headers: {
-            'Content-Type': 'application/json',
-            Authorization: `Bearer ${token}`,
-          },
+          // headers: {
+          //   'Content-Type': 'application/json',
+          //   Authorization: `Bearer ${token}`,
+          // },
```
Comment (edward-stalley-work) [inline client/src/pages/BusinessDivisions/BusinessDivisionsListAndEdit.tsx:124]: This is all commented out? Think this is the cause of update button not working and going to login.

File: client/src/pages/BusinessDivisions/BusinessDivisionsListAndEdit.tsx [hunk 8]
```diff
@@ -186,9 +186,9 @@ const BusinessDivisionsListAndEdit: React.FC = () => {
         try {
           const response = await axios.get('http://127.0.0.1:8000/api/master-business-divisions/', {
           // const response = await axios.get('http://54.178.202.58:8000/api/planningprojects/', {
-            headers: {
-              Authorization: `Bearer ${token}`, // Add token to request headers
-            },
+            // headers: {
+            //   Authorization: `Bearer ${token}`, // Add token to request headers
+            // },
```
Comment (edward-stalley-work) [inline client/src/pages/BusinessDivisions/BusinessDivisionsListAndEdit.tsx:191]: why is this commented out?
If its not needed please remove .

File: backend/api/views.py [hunk 9]
```diff
@@ -288,8 +288,25 @@ def put(self, request, *args, **kwargs):
 
         return Response({'message': 'Business divisions updated successfully'}, status=status.HTTP_200_OK)
 class MasterBusinessDivisionDestroy(generics.DestroyAPIView):
-    serializer_class = MasterBusinessDivisionSerializer
-    permission_classes = [IsAuthenticated]
+    # serializer_class = MasterBusinessDivisionSerializer
+    # permission_classes = [IsAuthenticated]
+    # lookup_field = 'pk'
```
Comment (edward-stalley-work) [inline backend/api/views.py:293]: why are these commented out?
If its not needed please remove .

File: backend/api/urls.py [hunk 10]
```diff
@@ -7,7 +7,7 @@
     path("master-business-division/<int:pk>", views.BusinessDivisionMasterRetrieve.as_view(), name="master-business-division-detail"),
     path("master-business-division/bulk-update/", views.MasterBusinessDivisionUpdate.as_view(), name="master-business-division-update"),
     # path("master-business-division/<int:pk>/update/", views.MasterBusinessDivisionUpdate.as_view(), name="master-business-division-update"), #temporary commented currently using bulk-update
```
Comment (edward-stalley-work) [inline backend/api/urls.py:9]: why is this commented out?
If its not needed please remove .

**Review by edward-stalley-work — CHANGES_REQUESTED** [review]
Please see comments

File: backend/api/urls.py (hunk 10 above)
Comment (ian-bpoc) [reply backend/api/urls.py]: already remove in my latest commit Ed

File: backend/api/views.py (hunk 9 above)
Comment (ian-bpoc) [reply backend/api/views.py]: already removed the commented code

File: client/src/pages/BusinessDivisions/BusinessDivisionsListAndEdit.tsx (hunk 8 above)
Comment (ian-bpoc) [reply client/src/pages/BusinessDivisions/BusinessDivisionsListAndEdit.tsx]: uncommented code

File: client/src/pages/BusinessDivisions/BusinessDivisionsListAndEdit.tsx (hunk 7 above)
Comment (ian-bpoc) [reply client/src/pages/BusinessDivisions/BusinessDivisionsListAndEdit.tsx]: done uncommented the code

**Pull request merged**